TomTomSDKExamples/Examples/UseCase/Offline/OfflineNavigationContent.swift

```diff
@@ -0,0 +1,689 @@
+        cameraUpdate.zoom = zoom
+        cameraUpdate.position = position
+        map?.applyCamera(cameraUpdate, animationDuration: animationDurationInSeconds)
+    }
+
+    func setCamera(trackingMode: TomTomSDKMapDisplay.CameraTrackingMode) {
+        map?.cameraTrackingMode = trackingMode
+
+        // Update chevron position on the screen so it is not hidden behind the navigation panel
+        if trackingMode == .followRoute || trackingMode == .follow {
+            let cameraUpdate = CameraUpdate(positionMarkerVerticalOffset: 0.4)
+            moveCamera(cameraUpdate: cameraUpdate)
+        }
+    }
+
+    func moveCamera(cameraUpdate: CameraUpdate) {
+        map?.moveCamera(cameraUpdate)
+    }
+}
+
+// MARK: TomTomSDKLocationProvider.LocationProviderObservable
+
+/// Extend OfflineMapCoordinator to conform to LocationProviderObservable by adding the following functions.
+///
+/// This extension enables the OfflineMapCoordinator to observe GPS updates and authorization changes.
+/// This means that when the application starts, the camera position and zoom level are updated in the onLocationUpdated callback function.
+/// The user then sees the current location.
+extension OfflineMapCoordinator: TomTomSDKLocationProvider.LocationProviderObservable {
+    func onLocationUpdated(location: GeoLocation) {
+        // Zoom and center the camera on the first location received.
+        animateCamera(zoom: 9.0, position: location.location.coordinate, animationDurationInSeconds: 1.5, onceOnly: true)
+    }
+
+    func onHeadingUpdate(newHeading _: CLHeading, lastLocation _: GeoLocation) {
+        // Handle heading updates
+    }
+
+    func onAuthorizationStatusChanged(isGranted _: Bool) {
+        // Handle authorization changes
+    }
+}
+
+// MARK: TomTomSDKMapDisplay.MapDelegate
+
+/// Extend OfflineMapCoordinator to conform to MapDelegate.
+///
+/// Update the OfflineMapCoordinator to plan a route and add it to the map after a long press.
+extension OfflineMapCoordinator: TomTomSDKMapDisplay.MapDelegate {
+    func map(_: TomTomMap, onInteraction interaction: MapInteraction) {
+        switch interaction {
+        case let .longPressed(coordinate):
+            offlineNavigationController.navigateToCoordinate(coordinate)
+        default:
+            // Handle other gestures
+            break
+        }
+    }
+
+    func map(_: TomTomMap, onCameraEvent event: CameraEvent) {
+        switch event {
+        case let .trackingModeChanged(mode):
+            // Handle camera tracking mode change
+            break
+        default:
+            break
+        }
+    }
+}
+
+/// Add an extension to OfflineNavigationController with a navigateToCoordinate function that plans a route, starts location simulation and starts the navigation process.
+///
+/// The call to the async planRoute function is wrapped in a Task so that it can be called from the navigateToCoordinate function, even though that function is not async.
```

Comment:
```suggestion
/// The call to the async `planRoute()` function is wrapped in a `Task` so that it can be called from the `navigateToCoordinate` function, even though that function is not async.
```

Reply:
👍 Done!